### PR TITLE
Fix food drain on movement.

### LINKED ISSUE #2044: Food drains too quickly.
Food appears to be mostly working, but it drains much more quickly than in the vanilla server.

### PR DESCRIPTION
**Pull request by LO1ZB**
Fixes #2044

### CODE REVIEW

File: src/Entities/Player.cpp [hunk 1]
```diff
@@ -2178,20 +2179,18 @@ void cPlayer::ApplyFoodExhaustionFromMovement()
...
-	if (GetWorld()->GetWorldAge() % 2 != 0)
-	{
-		return;
-	}
-
 	// Calculate the distance travelled, update the last pos:
-	Vector3d Movement(GetPosition() - m_LastPos);
-	Movement.y = 0;  // Only take XZ movement into account
+	double SpeedX = abs(m_Speed.x);
+	double SpeedZ = abs(m_Speed.z);
```
Comment (worktycho) [inline src/Entities/Player.cpp]: The abs is unnecessary.

Comment (SphinxC0re) [inline src/Entities/Player.cpp]: Why? Is the speed always positive?

File: src/Entities/Player.cpp [hunk 2]
```diff
@@ -2178,20 +2179,18 @@ void cPlayer::ApplyFoodExhaustionFromMovement()
...
-	Vector3d Movement(GetPosition() - m_LastPos);
-	Movement.y = 0;  // Only take XZ movement into account
+	double SpeedX = m_Speed.x;
+	double SpeedZ = m_Speed.z;
+	double BaseExhaustion(sqrt((SpeedX * SpeedX) + (SpeedZ * SpeedZ)));
 
 	// Apply the exhaustion based on distance travelled:
-	double BaseExhaustion = Movement.Length();
-	if (IsSprinting())
+	if (IsFlying() || IsClimbing())
```
Comment (SphinxC0re) [inline src/Entities/Player.cpp]: Are you sure, that no exhaustion is applied, when climbing up a ladder?

File: src/Entities/Player.cpp (hunk 1 above)
Comment (worktycho) [inline src/Entities/Player.cpp]: no, but a number squared is, which is all you use it for.

File: src/Entities/Player.cpp (hunk 2 above)
Comment (LO1ZB) [inline src/Entities/Player.cpp]: yes http://minecraft.gamepedia.com/Hunger#Exhaustion_level_increase

**Pull request merged**